src/NSubstitute/SubstituteExtensions.cs

```diff
@@ -354,5 +371,21 @@ private static ValueTask<T> CompletedValueTask<T>(T result)
         {
             return new ValueTask<T>(result);
         }
+
+        private static void ReThrowIfFaulted<T>(Task<T> task)
```

Comment:
Maybe `ReThrowOnSubstituteFault`? At present sounds like it will throw whenever faulted.

Reply:
Good catch, I also had doubts when named that method. Renamed to `ReThrowOnNSubstituteFault`. Is it conceptually fine?

Reply:
I think that's fine. Not that important for private scope, but I think it makes the calling methods read a bit clearer. 👍